### PR TITLE
INTLY-8400: Removed JobRunningTimeExceeded (>300) and include in upgrades

### PR DESCRIPTION
**Pull request by pepedocs**
Jira: https://issues.redhat.com/browse/INTLY-8400
https://issues.redhat.com/browse/CSSRE-806
https://issues.redhat.com/browse/CSSRE-1004

Summary:
1. Removed JobRunningTimeExceeded with the 300s threshold
2. Attempt to update "upgrade.template.yml" to include these changes for the next patch relase. I will need guidance on this as I'm not quite sure how this should be done properly. Basically, we need backup monitoring will be included in the next release. CC @david-martin . 
3. The fix for CronJobsFailed was already merged 2 months ago but was not in the 1.7 release, we need to include this in the next patch release.

Please read the Jira for more accurate and updated information. 

P.S. I base off my PR from the 1.7 branch however just found out I can only make a PR to the master branch. TBH I'm not quite sure about the correct process for this particular repo.

### CODE REVIEW

**Comment by pepedocs** [comment]
CC @alanmoran @david-martin

**Comment by austincunningham** [comment]
Thanks for your contribution @pepedocs 
Can you provide verification step to confirm your pr does what expected.

The pr seems to be removing the warning level alert for JobRunningTimeExceeded so this should be removed from elsewhere in the installer see https://github.com/integr8ly/installation/search?q=JobRunningTimeExceeded&unscoped_q=JobRunningTimeExceeded

**Comment by pepedocs** [comment]
> Thanks for your contribution @pepedocs
> Can you provide verification steps to confirm your pr does what expected.
> 
> The pr seems to be removing the warning level alert for JobRunningTimeExceeded so this should be removed from elsewhere in the installer see https://github.com/integr8ly/installation/search?q=JobRunningTimeExceeded&unscoped_q=JobRunningTimeExceeded

@austincunningham I've just removed the references to the JobRunningTimeExceeded warning alert.

In regards to the verification, this is really difficult to test. It requires a particular scenario where some metric data are delayed due to possibly prometheus app delays or federation of metric data. Please see the ticket here which is best explained - https://issues.redhat.com/browse/CSSRE-1004

Can the absence of this warning alert in the alertmanager/prometheus dashboard considered verification already? We just need to remove this warning alert as besides it's not accurate the 300s threshold is also not reasonable.

**Comment by jhellar** [comment]
For the verification, IMO it is enough to check that the alert is gone. This is how it will be also tested for 1.7.1 - https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases/-/blob/1.7.1/tests/alerts/c06-verify-that-there-is-only-one-alert-for-job-timeout.md

File: docs/monitoring_docs.asciidoc [hunk 1]
```diff
@@ -179,14 +179,6 @@ receivers:
   - send_resolved: true
     to: cssre-alerts@redhat.com
 - name: deadmansswitch
-inhibit_rules:
```
Comment (austincunningham) [inline docs/monitoring_docs.asciidoc:182]: only need to remove the target_match for warning
``` yaml
inhibit_rules:
- source_match:
    alertname: 'JobRunningTimeExceeded'
    severity: 'critical'
  equal: ['alertname', 'job', 'label_cronjob_name']
```

File: roles/middleware_monitoring_config/templates/alertmanager.yml.j2 [hunk 2]
```diff
@@ -51,13 +51,5 @@ receivers:
   webhook_configs:
     - url: "{{ dms_webhook_url }}"
 {% endif %}
-inhibit_rules:
```
Comment (austincunningham) [inline roles/middleware_monitoring_config/templates/alertmanager.yml.j2:54]: same here only need to remove the target_match for warning
``` yaml
inhibit_rules:
- source_match:
    alertname: 'JobRunningTimeExceeded'
    severity: 'critical'
  equal: ['alertname', 'job', 'label_cronjob_name']
```

File: docs/monitoring_docs.asciidoc (hunk 1 above)
Comment (pepedocs) [reply docs/monitoring_docs.asciidoc:182]: It looks like the source match will be useless without the target match, that's why I removed it. Please let me know if Im missing something here.

**Comment by pepedocs** [comment]
@austincunningham Could yoou also please verify if I made the right changes so that these alerts will be applied in the next upgrade?

**Comment by austincunningham** [comment]
@pepedocs deployed this to a cluster this morning and no `JobRunningTimeExceeded` alerts were showing up in Prometheus, going to deploy your latest changes to see if that resolves the issue

**Comment by pepedocs** [comment]
> @pepedocs deployed this to a cluster this morning and no `JobRunningTimeExceeded` alerts were showing up in Prometheus, going to deploy your latest changes to see if that resolves the issue

Thanks for testing it. I'm quite sure why the changes would cause all the Job* alerts to disappear. Please let me know when you have updates. If it fails your test, I will investigate on this tomorrow and test it on a live cluster.

**Comment by austincunningham** [comment]
@pepedocs second time was the same, decided to see if it was something I was doing wrong so tried master and it was the same , I was missing an install flag `-e backup_restore_install=true` , will test again with the updated flag.

**Comment by pepedocs** [comment]
> @pepedocs second time was the same, decided to see if it was something I was doing wrong so tried master and it was the same , I was missing an install flag `-e backup_restore_install=true` , will test again with the updated flag.

By the way could you also please check/test if the alert `CronJobsFailed` is using the metric `kube_job_failed` instead of `kube_job_status_failed`? This is part of the fix.

**Comment by pepedocs** [comment]
@austincunningham Thanks for testing this out again. Could someone in your team please provide guidance on this? The changes are pretty straightforward, I'm almost out of ideas why it removes the cron job alert. The behavior expected for these changes must be;

1. JobRunningTimeExceeded must exist but only for the 600s threshold.
2. CronJobsFailed alert must exist but should use the `kube_job_failed` prom metric.

CC @david-martin

**Comment by pepedocs** [comment]
@austincunningham Looking at the install logs you've provided, it seems like it didn't perform the upgrades for the backup monitoring. Particularly, it doesn't have the task `Delete existing alerts` (https://github.com/integr8ly/installation/blob/master/roles/backup/tasks/upgrade.yaml)

Therefore, can someone from your team please provide the steps on how/what files to update so that changes such as these backup monitoring alerts will be applied to upgrades and not just installs? This is not documented AFAICT in this repository.

**Comment by pepedocs** [comment]
@austincunningham I'm still not sure how to include these changes to integreatly upgrades (see my previous comments). However, I've just made another attempt (changes) and I've simply moved the backup monitoring upgrade role to the upgrade.yml playbook instead of the template.

**Could you please perform a test if you think this will fix the issue?** Reason why I attemtpted this is for us to save time waiting for the next day and most importantly because I also saw a PR that was modified the said playbook for including some changes in the next upgrades, so I'm really just inferring here.

**Comment by austincunningham** [comment]
/approve 
/lgtm
### Logs
https://issues.redhat.com/secure/attachment/12475971/12475971_integreatly-upgrade.log
https://issues.redhat.com/secure/attachment/12475972/12475972_integreatly-install.log

/assign @jessesarn

**Pull request merged**